### ISSUE TITLE
VTU doesn't show method exposed via context expose

### ISSUE THREAD

**Issue by novellac** [vuejs/vue-test-utils-next#1065]
### Subject of the issue

**tl;dr: Unable to unit test a method which I expose via context expose.**

I am using Vue 3.2, VTU 2.0.0-rc.16, Composition API, and JS (no TS).

I have two components, `Child` and `Parent`. `Child` returns a render function `h`. `Child` also exposes a method, `increment`, using [context expose](https://v3.vuejs.org/api/options-composition.html#setup). In `Parent`, I implement `Child` and call the `increment` method. All good so far in the browser. 
Now I am writing units for `Child`. Usually (using Composition or Options APIs), I would be able to call the method directly from the vm, and test the expected output. However, when `increment` is made available using expose, `increment` isn't available on the vm. How can I unit test the `increment` method in `Child`?

### Steps to reproduce

Example: See this [Github repo with simplified example](https://github.com/novellac/expose-units-example). Clone, serve, and run units locally. The unit test for the `Child` spec should fail, illustrating the issue.

1. Create two components, `Child` and `Parent`. 
1. In the `Child` setup method, use [context expose](https://v3.vuejs.org/api/options-composition.html#setup) to expose a method, `increment` which increments a ref. 
1. In `Parent`, call the `increment` method.
1. Create a unit test for `Child`. In the unit test, call `wrapper.vm.increment()` and expect that the output is incremented.
- Test will fail. See that the error reads `TypeError: wrapper.vm.increment is not a function`

### Expected behaviour

I would expect that the method `increment` would be available on the vm.

### Actual behaviour

The method is not available on the vm.

**Comment by cexbrayat** [comment]
Hi @novellac ,

Thank you for opening an issue, and providing a repro.
I took a quick look, as I recently worked on the `expose` topic in VTU-next.

TL;DR: The render function is the reason it does not work here. I opened a PR to fix it.


Let me explain.
Using sfc.vuejs.org, you can see that a component like:

```vue
<template>
  <div>{{ other }}</div>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue'

export default defineComponent({
  name: 'Hello',

  setup(props, { expose }) {
    const other = ref('other')
    expose({ other })
    return {
      other
    }
  }
})
</script>
```

generates JS like this when compiled:
```js
import { defineComponent, ref } from 'vue'

const __sfc__ = defineComponent({
  name: 'Hello',

  setup(props, { expose }) {
    const other = ref('other')
    expose({ other })
    return {
      other
    }
  }
})

import { toDisplayString as _toDisplayString, openBlock as _openBlock, createElementBlock as _createElementBlock } from "vue"
function render(_ctx, _cache, $props, $setup, $data, $options) {
  return (_openBlock(), _createElementBlock("div", null, _toDisplayString(_ctx.other), 1 /* TEXT */))
}
__sfc__.render = render
__sfc__.__file = "App.vue"
export default __sfc__
```

Note the existence of `ctx` containing the `other` property.
We use this `ctx` in VTU as the `vm` (`ctx` is `vm.$.proxy`).
So in a component with a template, you can access `wrapper.vm.other` in your test.

Now, if you write the same component with a render function:
```vue
<script lang="ts">
import { defineComponent, h, ref } from 'vue'

export default defineComponent({
  name: 'Hello',

  setup(props, { expose }) {
    const other = ref('other')
    expose({ other })
    return () => h('div', other.value)
  }
})
</script>
```

then the compiler outputs:
```js
/* Analyzed bindings: {} */

import { defineComponent, h, ref } from 'vue'

const __sfc__ = defineComponent({
  name: 'Hello',

  setup(props, { expose }) {
    const other = ref('other')
    expose({ other })
    return () => h('div', other.value)
  }
})

__sfc__.__file = "Comp.vue"
export default __sfc__
```

No `ctx`, so `vm.$.proxy` is empty in VTU-next, hence your issue.

The behavior difference can be seen in vue-next [here](https://github.com/vuejs/vue-next/blob/d19cfc0503dddbded5e599bef0965aba3d25ca28/packages/runtime-core/src/component.ts#L674)

The proposed fix is a bit naive, but returns `vm` instead of `vm.$.proxy` in the case that the component returned a render function in its setup, and should fix your use-case.

**Comment by novellac** [comment]
@cexbrayat @lmiller1990 Thank you for the thorough explanation, and for your work on it. Much appreciated!
